rama-tcp/src/client/connect.rs

```diff
@@ -154,8 +166,8 @@ async fn tcp_connect_inner<State, Dns, Connector>(
     ctx: &Context<State>,
     domain: Domain,
     port: u16,
-    dns: Dns,
-    connector: Connector,
+    dns: DnsResolveIpMode,
```

Comment:
Does this still even compile? You remove `Dns` but `DnsResolveIpMode` doesn't provide actual dns functionality, yet you call `lookup_ipv4` later on it...

shouldn't this instead be `dns_mode: DnsResolveIpMode` while keeping `dns: Dns`?